haystack/document_stores/weaviate.py

```diff
@@ -697,17 +697,31 @@ def _get_all_documents_in_index(
                 query = query.with_where(filter_dict)
 
             if all_docs:
-                # .with_limit() must be used with .with_offset, of the latter won't work properly
-                #   https://weaviate-python-client.readthedocs.io/en/latest/weaviate.gql.html?highlight=offset#weaviate.gql.get.GetBuilder.with_offset
+                # Passing offset:0 raises an error, so we pass it only after the first round
+                # `.with_limit()` must be used with `.with_offset`, or the latter won't work properly
+                # https://weaviate-python-client.readthedocs.io/en/latest/weaviate.gql.html?highlight=offset#weaviate.gql.get.GetBuilder.with_offset
                 query = query.with_limit(100).with_offset(offset=len(all_docs))
 
-            result = query.do()
+            try:
+                result = query.do()
+            except Exception as e:
+                raise ValueError(f"Weaviate raised an exception: {e}")
+
+            if "errors" in result:
+                raise ValueError(f"Query results contain errors: {result.get('errors')}")
```

Comment:
A curiosity here: I see that often in the code you're doing `result.get("errors")` after checking the existence of the key. I usually use `dict.get(key, default)` when I'm assuming the key might not exist, and `dict[key]` when it must exist. Is this a precise choice, or just style?

Reply:
Uhm I think I did it only once? In this case it was just an overlook, your assumption about using brackets or `get()` is correct!

Reply:
Alright "often" was definitely an overstatement :smile: I saw it on line 720 as well